src/Streamly/Coreutils/Uniq.hs

```diff
@@ -0,0 +1,161 @@
+module Streamly.Coreutils.Uniq (
+      ignCase
+    , splitOnNewLine
+    , uniqCount
+    , uniqRepeated
+    , uniqDistinct
+    , defaultUniqOptions
+    , UniqOptions (..)
+    , skpField
+    , process
+    , comp
+    , select
+   )
+where
+
+import Streamly.Coreutils.Common
+import qualified Streamly.Prelude as S
+import qualified Streamly.Memory.Array as A
+import qualified Streamly.Internal.Data.Fold as FL
+import qualified Streamly.Data.Unicode.Stream as Un
+import qualified Streamly.Internal.FileSystem.File as File
+
+import Data.Word (Word8)
+import Data.Char (toLower)
+import System.IO (Handle, stdout)
+import System.IO.Unsafe (unsafePerformIO)
+import Foreign.Storable (Storable)
+import Control.Monad.IO.Class (MonadIO)
+import Streamly.Internal.Data.Stream.StreamK.Type (IsStream)
+
+-------------------------------------------------------------------------------
+-- Record for options used with uniq
+-------------------------------------------------------------------------------
+
+{-# INLINE intMax #-}
+
+intMax :: Int
+intMax = maxBound
+
+
+data UniqOptions = UniqOptions {
+                       count :: Bool
+                     , repeated :: Bool       -- display only duplicate lines once for each group
+                     , duplicate :: Bool      -- print all duplicate lines
+                     , skipFields :: Int      -- skips fields, then chars & checks remaining
+                     , skipChar :: Int
+                     , checkChar :: Int
+                     , ignoreCase :: Bool
+                     , unique :: Bool         -- print only unique lines
+                     , zeroTerminated :: Bool
+                  }
+
+defaultUniqOptions :: UniqOptions
+defaultUniqOptions = UniqOptions True False False 0 0 intMax False True False
+
+
+-------------------------------------------------------------------------------
+-- helper functions for uniq
+-------------------------------------------------------------------------------
+
+
+-- convert stream of bytes to stream of characters - ignore case if bool is True
+
+ignCase :: (IsStream t, Monad m) => Bool -> t m Word8 -> t m Char
+ignCase flag strm = if flag == True
+                    then S.map toLower $ Un.decodeLatin1 strm
+                    else Un.decodeLatin1 strm
+
+
+-- convert stream of char to stream of Array Char by breaking on newline
+
+splitOnNewLine :: (MonadIO m, IsStream t, Monad m) => t m Char -> t m (A.Array Char)
+splitOnNewLine strm = S.splitOnSuffix (== '\n') A.write strm
+
+
+isSpace :: Char -> Bool
+isSpace c = c == ' '
+
+safeLast :: [a] -> Maybe a
+safeLast [] = Nothing
+safeLast (x:[]) = Just x
+safeLast (x:xl) = safeLast xl
+
+
+mergeStringChar :: String -> Char -> String
```

Comment:
This is rather inefficient. What are you trying to do here? Maybe there is a better way.

Reply:
I want to fold a `t m Char` into `t m String`. Basically, want to do this
`[' ', ' ', 'a', 'b', 'c' ,' ', ' ', 'e']  -> ["  ", "abc", "  ", "e"]` using streams

Reply:
Then drop a few strings from the stream generated (based on options provided) and then convert the `String` stream back to a `Char` stream.

Reply:
```
groupCS :: (Monad m, IsStream t) => t m Char -> t m String
groupCS = S.groupsBy xorCS FL.toList
  where
    xorCS a b = (isSpace a && isSpace b) || (not (isSpace a || isSpace b))
```
Use this instead.

Reply:
I need multiple spaces. :+1: